homeassistant/components/kodi/media_player.py

```diff
@@ -435,6 +439,26 @@ async def ws_loop_wrapper():
         # run until the websocket connection is closed.
         self.hass.loop.create_task(ws_loop_wrapper())
 
+    async def async_added_to_hass(self):
+        """Connect the websocket if needed."""
+        if not self._enable_websocket:
+            return
+
+        await self._async_connect_websocket_if_disconnected()
+
+        self.async_on_remove(
+            async_track_time_interval(
+                self.hass,
+                self._async_connect_websocket_if_disconnected,
+                WEBSOCKET_WATCHDOG_INTERVAL,
+            )
+        )
+
+    async def _async_connect_websocket_if_disconnected(self, *_):
+        """Reconnect the websocket if it fails."""
+        if not self._ws_server.connected:
+            await self.hass.async_create_task(self.async_ws_connect())
```

Comment:
Why should we create a task here instead of awaiting the ws connect task? If we don't want to wait in `async_added_to_hass` I'd move the task creation to line 447. There's already a task created inside `async_track_time_interval`.

Reply:
I'll change it call directly and then change `async_added_to_hass` to use `hass.loop.create_task` so we know the task won't block startup.